peekingduck/cli.py

```diff
@@ -149,13 +104,111 @@ def run(config_path: str, node_config: str) -> None:
+        "the command is invoked."
+    ),
+    required=False,
+)
+@click.option(
+    "--node_type",
+    help=(
+        "Node type, only accepts values from existing node types defined in "
+        f"{PEEKINGDUCK_NODE_TYPES}."
+    ),
+    required=False,
+)
+@click.option(
+    "--node_name",
+    help=(
+        "\b\nName of new custom node. The name cannot be a duplicate of an \n"
+        "existing custom node. The name has the following requirements:\n"
+        "- Minimum 2 characters\n"
+        "- Can only contain alphanumeric characters, dashes and underscores \n"
+        "  /[[a-zA-Z0-9_\\-]/\n"
+        "- Must start with an alphabet\n"
+        "- Must end with an alphanumeric character"
+    ),
+    required=False,
+)
+def create_node(
+    node_subdir: Optional[str] = None,
+    node_type: Optional[str] = None,
+    node_name: Optional[str] = None,
+) -> None:
+    """Automates the creation of a new custom node.
+
+    If the options `node_subdir`, `node_type`, or `node_name` are not
+    specified, users will be prompted them for the values while performing
+    checks to ensure value validity.
+    """
+    click.secho("Creating new custom node...")
+    project_dir = Path.cwd()
+    node_type_choices = click.Choice(PEEKINGDUCK_NODE_TYPES)
+
+    node_subdir = _verify_option(node_subdir, value_proc=_ensure_relative_path)
+    if node_subdir is None:
+        node_subdir = click.prompt(
+            f"Enter node directory relative to {project_dir}",
+            default="src/custom_nodes",
+            value_proc=_ensure_relative_path,
+        )
+    node_dir = project_dir / node_subdir
+
+    node_type = _verify_option(
+        node_type, value_proc=click.types.convert_type(node_type_choices)
+    )
+    if node_type is None:
+        node_type = click.prompt("Select node type", type=node_type_choices)
+
+    partial_ensure_valid_name = functools.partial(
+        _ensure_valid_name, node_dir, node_type
+    )
+    node_name = _verify_option(node_name, value_proc=partial_ensure_valid_name)
+    if node_name is None:
+        node_name = click.prompt(
+            "Enter node name",
+            default="my_custom_node",
+            value_proc=partial_ensure_valid_name,
+        )
+
+    config_path = node_dir / "configs" / node_type / f"{node_name}.yml"
+    script_path = node_dir / node_type / f"{node_name}.py"
+    click.echo(f"\nNode directory:\t{node_dir}")
+    click.echo(f"Node type:\t{node_type}")
+    click.echo(f"Node name:\t{node_name}")
+    click.echo("\nCreating the following files:")
+    click.echo(f"\tConfig file: {config_path}")
+    click.echo(f"\tScript file: {script_path}")
+
+    proceed = click.confirm("Proceed?", default=True)
+    if proceed:
+        config_path.parent.mkdir(parents=True, exist_ok=True)
+        script_path.parent.mkdir(parents=True, exist_ok=True)
+        pkd_dir = Path(__file__).resolve().parent
+        template_name = "node_template"
```

Comment:
The AISG license header would be automatically created as it's in node_template, but I feel that it may put off users as this is their own code and the license is entirely up to them. We could either:
1. Exclude copying over the first few lines with the license header
2. Remove the license header from the template entirely

What do you think?

Reply:
Sure, since we are keeping a pretty standard python script structure
```
<license>

<module docstring>

<code>
```
I can just copy from the first line that starts with `"`. We can still keep the license in the template file since I think there's some checks for licenses in all script files in our CI.

Reply:
Implemented https://github.com/aimakerspace/PeekingDuck/pull/525/commits/d3dd97e598d110381adf4546e235fb004cbcbc52

Implemented a fail-safe to just copy the entire template file if a line starting with `"` could not be found.